### ISSUE TITLE
ping unreachable node sometimes times out >1min

### ISSUE THREAD

**Issue by alanshaw** [ipfs/go-ipfs#5799]
Sometimes this takes ~10s which is [consistent with the code](https://github.com/ipfs/go-ipfs/blob/31099e882469fab12a98f2069e38c847f0b1188c/core/commands/ping.go#L22) but it frequently takes >1 minute in CI:

<img width="1362" alt="screenshot 2018-11-27 at 13 37 41" src="https://user-images.githubusercontent.com/152863/49085320-b059a700-f249-11e8-910b-d6442d22d3d5.png">

**Comment by magik6k** [comment]
The real timeout is `numPings*pingTimeout` - 
https://github.com/ipfs/go-ipfs/blob/31099e882469fab12a98f2069e38c847f0b1188c/core/commands/ping.go#L106

What is `-n` set to in those tests?

**Comment by alanshaw** [comment]
2 in that test https://github.com/ipfs/interface-ipfs-core/blob/0ca86fa081edfeb69be83d2a23fe2af71bfafde7/js/src/ping/ping.js#L54-L62

**Comment by alanshaw** [comment]
I thought that because the test is for an unreachable node it would actually drop in here https://github.com/ipfs/go-ipfs/blob/31099e882469fab12a98f2069e38c847f0b1188c/core/commands/ping.go#L90 and so the timeout would be 10s - which is consistent with the what we're seeing some of the time...

**Comment by Stebalien** [comment]
Triage:  @alanshaw are you still seeing this?

>  and so the timeout would be 10s - which is consistent with the what we're seeing some of the time...

That's the timeout for connecting to the peer. There's another timeout below that's `kPingTimeout * numPings`. So this should total to `kPingTimeout * (numPings+1)` which should be 30s in your case.

So yeah, something is broken.

**Comment by alanshaw** [comment]
I've not removed the additional time we've allocated for the test to pass ;)

Looking at recent CI we've not seen it for a while.

IMHO we should close this as multiple releases have happened since reporting, we've switched CI and now use the "test" profile in tests - any one of those factors (or something completely different) could have fixed this.

I can open a new issue if I notice it again.

> That's the timeout for connecting to the peer. There's another timeout below...

Just FYI, I was not expecting it to get any further since this is for pinging an unreachable node. The Peer ID will not be in the peer store.
